src/Tools/BuildValidator/BuildConstructor.cs

```diff
@@ -48,64 +51,125 @@ public Compilation CreateCompilation(MetadataReader metadataReader, string name)
             {
                 var compilation = language switch
                 {
-                    LanguageNames.CSharp => CreateCSharpCompilation(pdbReader, name),
-                    LanguageNames.VisualBasic => CreateVisualBasicCompilation(pdbReader, name),
-                    _ => throw new InvalidDataException($"{language} is not a known language")
+                    LanguageNames.CSharp => CreateCSharpCompilationAsync(optionsReader, name),
+                    LanguageNames.VisualBasic => CreateVisualBasicCompilationAsync(optionsReader, name),
+                    _ => Task.FromException<Compilation>(new InvalidDataException($"{language} is not a known language"))
                 };
 
                 return compilation;
             }
 
-            throw new InvalidDataException("Did not find language in compilation options");
+            return Task.FromException<Compilation>(new InvalidDataException("Did not find language in compilation options"));
+        }
+
+        private ImmutableArray<MetadataReferenceInfo> GetMetadataReferenceInfos(CompilationOptionsReader compilationOptionsReader)
+        {
+            return compilationOptionsReader.GetMetadataReferences();
+        }
```

Comment:
Why not call `compilationOptionsReader.GetMetadataReferences()` directly in callers of this method and remove the method completely?

Reply:
Fixed